### PR TITLE
Update `precommit:emoji` script after GitHub's sunset of SVN

### PR DESCRIPTION
**Pull request by desrosj**
All build script related changes form #5804 have been copied over.

Trac ticket: https://core.trac.wordpress.org/ticket/60520

---
**This Pull Request is for code review only. Please keep all other discussion in the Trac ticket. Do not merge this Pull Request. See [GitHub Pull Requests for Code Review](https://make.wordpress.org/core/handbook/contribute/git/github-pull-requests-for-code-review/) in the Core Handbook for more details.**

### CODE REVIEW

**Comment by kraftbj** [comment]
Tested. Only thing I ran into was permission needing updating on the shell script. Submitted a PR to add that via https://github.com/desrosj/wordpress-develop/pull/153

**Comment by desrosj** [comment]
Thanks @kraftbj! I've merged that.

To bring a brief summary here, here are a few things I don't love about the current implementation (though not strongly opposed to):
- I don't love adding `gh` as a requirement for the script.
- I would prefer that the necessary script be included in `Grunt.js` instead of a separate directory for shell scripts (where this will be the only one).

**Comment by kraftbj** [comment]
I don't love adding `gh` either, but given that we _do_ use GitHub as a partner service extensively already, it seems okay for now to get this out the door.

I tried to make it all part of the Grunt JS file in https://github.com/desrosj/wordpress-develop/pull/154 -- can you give that a spin?

File: Gruntfile.js [hunk 1]
```diff
@@ -332,7 +332,6 @@ module.exports = function(grunt) {
 					[ WORKING_DIR + 'wp-admin/js/tags-suggest.js' ]: [ './src/js/_enqueues/admin/tags-suggest.js' ],
 					[ WORKING_DIR + 'wp-admin/js/tags.js' ]: [ './src/js/_enqueues/admin/tags.js' ],
 					[ WORKING_DIR + 'wp-admin/js/site-health.js' ]: [ './src/js/_enqueues/admin/site-health.js' ],
-					[ WORKING_DIR + 'wp-admin/js/site-icon.js' ]: [ './src/js/_enqueues/admin/site-icon.js' ],
```
Comment (peterwilsoncc) [inline Gruntfile.js]: Is this intentional, it looks unrelated.

**Comment by desrosj** [comment]
Thanks all for keeping this going! I think we're at a good spot. There's just one outstanding question about what seems like an unintentional deleted line.

**Comment by kraftbj** [comment]
I concur. If the deletion was intentional (I'm not sure why it would be), it should be noted in the commit to explain the connection. But, with that line restored, LGTM.

**Comment by desrosj** [comment]
It was me that removed that line, wasn't it? Just realized that looking through this again. 🙃

Since I can't recall why I included that, or see a reason how it's related, I've gone and removed it.

I also added a new step to the Actions workflow that tests our build process to run the `precommit:emoji` script to confirm all expected changes are included. GitHub is not syncing pushes right now for some reason. But once I confirm it correctly flags missing changes, I'll get this merged.

**Comment by peterwilsoncc** [comment]
> I also added a new step to the Actions workflow that tests our build process to run the precommit:emoji script to confirm all expected changes are included. GitHub is not syncing pushes right now for some reason. But once I confirm it correctly flags missing changes, I'll get this merged.

As the script currently reads the files from the `main` branch, this may need to use something version specific so it doesn't trigger build failures due to modified files as the folder is updated upstream.

File: .github/workflows/test-build-processes.yml [hunk 2]
```diff
@@ -31,7 +31,7 @@ jobs:
   # Tests the WordPress Core build process on multiple operating systems.
   test-core-build-process:
     name: Core running from ${{ matrix.directory }}
-    uses: WordPress/wordpress-develop/.github/workflows/callable-test-core-build-process.yml@trunk
```
Comment (swissspidy) [inline .github/workflows/test-build-processes.yml:34]: To undo before commit

**Comment by desrosj** [comment]
> As the script currently reads the files from the `main` branch, this may need to use something version specific so it doesn't trigger build failures due to modified files as the folder is updated upstream.

I've tested using a tag reference instead of branch, and it seems to work alright. Talked through this a bit in Slack with @peterwilsoncc and since this adds an additional spot to update when updating the library, it would be nice to automate it a bit.

**Comment by desrosj** [comment]
Merged in https://core.trac.wordpress.org/changeset/57758.
